rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -84,18 +87,33 @@ NodeBase::NodeBase(
   }
 
   // Create the rcl node and store it in a shared_ptr with a custom destructor.
-  rcl_node_t * rcl_node = new rcl_node_t(rcl_get_zero_initialized_node());
+  std::unique_ptr<rcl_node_t> rcl_node(new rcl_node_t(rcl_get_zero_initialized_node()));
```

Comment:
I changed this only to get rid of false positive "double free" error from cppcheck. It didn't like that it saw two `delete rcl_node;` statements: one in the error handling after calling `rcl_parse_arguments()`, and another in the error handling of `rcl_node_init()`.